src/blob.jl

```diff
@@ -61,11 +61,28 @@ function Base.:-(blob1::Blob, blob2::Blob)
     getfield(blob1, :offset) - getfield(blob2, :offset)
 end
 
-@noinline function _throw_assert_not_null_error(typename::Symbol)
-    throw(AssertionError("Null pointer dereference in Blob{$(typename)}"))
+@eval @inline BOUNDSCHECK_ON_DEREF_ENABLED() = false
+
```

Comment:
Also, does making the default `false` make this a breaking API change (in theory `blob[]` could now segfault instead of throwing an exception?)

probably a @NHDaly question

Reply:
I think this is a non-breaking change. You are allowed to change the behavior in error-conditions as part of a non-breaking release, I'm pretty sure. It _should_ probably be a minor version bump at least though.